sky/cli.py

```diff
@@ -3077,7 +3077,7 @@ def spot_launch(
     if name is None:
         name = backend_utils.generate_cluster_name()
     else:
-        backend_utils.check_cluster_name_is_valid(name)
+        clouds.Cloud.check_cluster_name_is_valid(name)
```

Comment:
Maybe comment that this does a basic regex check, while name length check will be done when the controller has decided which cloud to use and begins provisioning the spot cluster?